### PR TITLE
[9.0][MIG] account invoice pricelist: Migration to v9

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @tecnativa

### CODE REVIEW

File: account_invoice_pricelist/__openerp__.py [hunk 1]
```diff
@@ -1,31 +1,14 @@
...
 
 {
     'name': 'Account - Pricelist on Invoices',
-    'version': '8.0.1.0.0',
+    'version': '9.0.1.0.0',
     'summary': 'Add partner pricelist on invoices',
     'category': 'Accounting & Finance',
-    'author': 'GRAP,Odoo Community Association (OCA)',
+    'author': 'GRAP,'
+              'Therp BV,'
```
Comment (pedrobaeza) [inline account_invoice_pricelist/__openerp__.py]: Add Tecnativa

File: account_invoice_pricelist/i18n/es.po [hunk 2]
```diff
@@ -26,7 +27,7 @@ msgstr "Factura"
 #: view:account.invoice:account_invoice_pricelist.view_account_invoice_search
 #: field:account.invoice,pricelist_id:0
 msgid "Pricelist"
-msgstr ""
+msgstr "rifa"
```
Comment (pedrobaeza) [inline account_invoice_pricelist/i18n/es.po]: Tarifa

File: account_invoice_pricelist/model/account_invoice.py [hunk 3]
```diff
@@ -1,58 +1,29 @@
...
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
 
-    # Column Section
     pricelist_id = fields.Many2one(
         comodel_name='product.pricelist', string='Pricelist',
         help="The pricelist of the partner, when the invoice is created"
         " or the partner has changed. This is a technical field used"
-        " for reporting.")
+        " for reporting.",
```
Comment (pedrobaeza) [inline account_invoice_pricelist/model/account_invoice.py]: Remove last part of the help, as this is not true anymore (it's not a technical field).

File: account_invoice_pricelist/model/account_invoice_line.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+from openerp import models, api
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.onchange('product_id')
+    def _onchange_product_id_account_invoice_pricelist(self):
+        if self.invoice_id.pricelist_id and self.invoice_id.partner_id:
```
Comment (pedrobaeza) [inline account_invoice_pricelist/model/account_invoice_line.py]: For avoiding a new indentation level, better to do:
```
if not self.invoice_id.pricelist_id or not self.invoice_id.partner_id:
    return
product = ...
```

File: account_invoice_pricelist/view/view.xml [hunk 5]
```diff
@@ -1,58 +1,48 @@
...
-                    <field name="pricelist_id"/>
-                </field>
+<odoo>
+
+    <record id="view_account_invoice_customer_form" model="ir.ui.view">
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_form"/>
+        <field name="arch" type="xml">
+            <field name="state" position="before">
+                <button type="object" name="button_update_prices_from_pricelist" string="Update prices from pricelist" attrs="{'invisible': ['|', ('pricelist_id', '=', False), ('state', 'not in', ['draft'])]}" />
```
Comment (pedrobaeza) [inline account_invoice_pricelist/view/view.xml]: I would put this under the pricelist field with a `class="oe_link"` attribute

File: account_invoice_pricelist/view/view.xml [hunk 6]
```diff
@@ -1,58 +1,48 @@
...
-        </record>
-
-    </data>
-</openerp>
-
-
+        </field>
+    </record>
+
+    <record id="view_account_invoice_supplier_form" model="ir.ui.view">
```
Comment (pedrobaeza) [inline account_invoice_pricelist/view/view.xml]: Remove this view, as supplier pricelist are not in v9.

File: account_invoice_pricelist/view/view.xml [hunk 7]
```diff
@@ -1,58 +1,48 @@
...
+                <attribute name="context">{'partner_id': partner_id, 'price_type': context.get('price_type') or False, 'type': type, 'pricelist_id': pricelist_id}</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_account_invoice_search" model="ir.ui.view">
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.view_account_invoice_filter"/>
+        <field name="arch" type="xml">
+            <xpath expr="//filter[@name='group_by_partner_id']" position="after">
```
Comment (pedrobaeza) [inline account_invoice_pricelist/view/view.xml]: If this view is shared for both invoice types, we should hide it for supplier ones. For sure there's a context variable that allows this.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done

File: account_invoice_pricelist/i18n/es.po [hunk 8]
```diff
@@ -26,7 +27,7 @@ msgstr "Factura"
 #: view:account.invoice:account_invoice_pricelist.view_account_invoice_search
 #: field:account.invoice,pricelist_id:0
 msgid "Pricelist"
-msgstr ""
+msgstr "Tarifa"
 
 #. module: account_invoice_pricelist
 #: help:account.invoice,pricelist_id:0
```
Comment (pedrobaeza) [inline account_invoice_pricelist/i18n/es.po]: This translation should change also as you have changed the help text.

File: account_invoice_pricelist/i18n/es.po [hunk 9]
```diff
@@ -26,7 +27,7 @@ msgstr "Factura"
 #: view:account.invoice:account_invoice_pricelist.view_account_invoice_search
 #: field:account.invoice,pricelist_id:0
```
Comment (pedrobaeza) [inline account_invoice_pricelist/i18n/es.po]: Translation definition doesn't work as in v8, so this for sure won't work as translation. Please export again and retranslate.

**Review by pedrobaeza — APPROVED** [review]
Minor comments, but :+1:

File: account_invoice_pricelist/i18n/es.po (hunk 9 above)
Comment (sergio-teruel) [reply account_invoice_pricelist/i18n/es.po]: Delete all translations??

Comment (pedrobaeza) [reply account_invoice_pricelist/i18n/es.po]: Well, code messages translations are kept the same, so they will be automatically purged on Transifex load, but at least leave Spanish one 100% translated.

File: account_invoice_pricelist/tests/test_account_invoice_pricelist.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        self.a_receivable = self.env["account.account"].create({
+            "name": "Test receivable account",
+            "code": "TEST_RA",
+            "user_type_id": self.at_receivable.id,
+            "reconcile": True,
+        })
+        self.partner = self.env['res.partner'].create({
+            'name': 'Test Partner',
+            'property_product_pricelist': 1,
+            'property_account_receivible_id': self.a_receivable.id,
```
Comment (NL66278) [inline account_invoice_pricelist/tests/test_account_invoice_pricelist.py]: typo: receivable not receivible

**Review by NL66278 — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@sergio-teruel please change what Ronald commented.

**Review by luismontalba — APPROVED** [review]
Functionality tested on runbot

**Comment by sergio-teruel** [comment]
@pedrobaeza @rafaelbn Changes done. @NL66278 Sorry I had lost my mind...

**Review by chienandalu — COMMENTED** [review]
The invoice doesn't inherit the pricelist from the Sale Order. Is that normal?

**Comment by pedrobaeza** [comment]
What do you mean, David?

**Comment by chienandalu** [comment]
@pedrobaeza Following these steps:

1. I make a sale order with a pricelist set.
2. I confirm that order and its shipments.
3. I create the invoice for that order: in this invoice no pricelist is assigned.

I thought the resultant invoice would be set with its parent sale order pricelist. Is that the expected behavior?

**Comment by pedrobaeza** [comment]
Well, that's something not covered in the module, as this one doesn't depend on sale nor purchase. For that, you would need to add glue modules account_invoice_pricelist_sale or account_invoice_pricelist_purchase. What the invoicing process guarantee is that the sales order prices are transferred to invoices.

**Review by chienandalu — APPROVED** [review]
then functionality tested on runbot is ok

**Comment by NL66278** [comment]
Just to remark that the functionality @chienandalu asked about, taking invoice pricelist from saleorder, if there is a sale-order, is part of this pull request for 8.0: https://github.com/OCA/account-invoicing/pull/215 It would be easy to migrate this to 9.0 to.

**Pull request merged**